src/polynomial.cpp

```diff
@@ -0,0 +1,35 @@
+#include "polynomial.h"
+#include "add.h"
+
+namespace SymEngine {
+
+    Polynomial::Polynomial(const std::string& var, map_uint_mpz&& dict) :
```

Comment:
Having numbers sorted usually speeds up computations. You might want to look at [this](http://stackoverflow.com/questions/2196995/is-there-any-advantage-of-using-map-over-unordered-map-in-case-of-trivial-keys)